### PR TITLE
[Feature] Button soft disabled

### PR DESCRIPTION
**Pull request by aappoalander**
## Description
This PR adds support for aria-disabled attribute and allows focusing disabled buttons while navigating with tab-button. In addition, this PR removes unstyled variant and support for Aria live. Components level implementation is also removed as it became redundant.

## Motivation and Context
For example, Submit buttons should be completely focusable when disabled to better communicate reasons behind disabled state for screen reader users. Aria-assertive / live does not work well for these use cases since it announces changes in the disabled state immediately even if focus has never reached the button it self. A better strategy for this is to use aria-describedby with aria-disabled. This way all relevant information is always available for screen reader users as well.

## How Has This Been Tested?
Tested with Styleguidist, CRA-TS and SSR test app using Chrome, Firefox, Edge, Safari with VoiceOver on MacOS.

## Release notes
- Breaking changes ⚒️ : support for Aria-disabled, unstyled variant removed along with mouseNoFocus, ariaAssertiveDisabled and ariaAssertiveUnDisabled attributes

### CODE REVIEW

**Comment by ketsappi** [comment]
Maybe add a note to PR description that Button is now removed from the components level.

File: src/core/Button/Button.md [hunk 1]
```diff
@@ -16,6 +16,24 @@ import { Button } from 'suomifi-ui-components';
...
+<>
+  <Button
+    aria-describedby="additiona-info"
+    aria-disabled={true}
+    onClick={() => console.log('Test button click')}
+    type="submit"
+  >
+    Aria-disabled button
+  </Button>
+  <Paragraph id="additiona-info">
```
Comment (ketsappi) [inline src/core/Button/Button.md:31]: Not a problem, but as this is a example we might want to fix the typo in the id;
From _additiona-info_ to _additiona**l**-info_

Comment (aappoalander) [reply src/core/Button/Button.md:31]: I'd also like to apply a style here, but no comments from design yet.

File: src/core/Button/Button.test.tsx [hunk 2]
```diff
@@ -1,44 +1,84 @@
...
-    <Button data-testid="button" disabled={true}>
-      Test
-    </Button>,
-  );
-  const { getByTestId, rerender } = button;
-  expect(getByTestId('button')).toBeDisabled();
+describe('Disabled Button', () => {
+  it('is disabled and changes to enabled', () => {
+    const button = render(<Button disabled={true}>Test</Button>);
+    const { getByRole, rerender } = button;
```
Comment (ketsappi) [inline src/core/Button/Button.test.tsx:32]: Could this and other tests be with "one line"?

```tsx
    const { getByRole, rerender } = render( <Button disabled={true}>Test</Button>);
```

File: src/core/Button/Button.tsx [hunk 3]
```diff
@@ -1,34 +1,44 @@
...
-import { UnstyledButton } from './UnstyledButton';
 
 type ButtonVariant =
   | 'default'
   | 'inverted'
   | 'secondary'
   | 'secondary-noborder'
-  | 'tertiary'
-  | 'unstyled';
-
```
Comment (ketsappi) [inline src/core/Button/Button.tsx:21]: Just for readability, might be good to leave the empty row between `type` and `export interface` here.

File: src/core/Button/Button.tsx [hunk 4]
```diff
@@ -41,102 +51,111 @@ export interface ButtonProps extends CompButtonProps, TokensProp {
...
+      children,
       ...passProps
     } = this.props;
-    const { variant } = passProps;
     const { className: iconPropsClassName, ...passIconProps } = iconProps;
-
-    if (variant === 'unstyled') {
-      return <UnstyledButton {...passProps} />;
-    }
+    const onClickProps = !!disabled || !!ariaDisabled ? {} : { onClick };
```
Comment (ketsappi) [inline src/core/Button/Button.tsx:84]: Maybe we could use singular, as there is only one property, --> `const onClickProp`

But not sure which is best, as this object is then spread later when used for button.

Comment (aappoalander) [reply src/core/Button/Button.tsx:84]: Prop should be sufficient for this case I guess

File: src/core/Button/Button.md (hunk 1 above)
Comment (aappoalander) [reply src/core/Button/Button.md:31]: Typo fixed and text for additional-info improved.

File: src/core/Button/Button.test.tsx (hunk 2 above)
Comment (aappoalander) [reply src/core/Button/Button.test.tsx:32]: Fixed. Also helped with quite a few naming issues, no name required for those any more.

File: src/core/Button/Button.tsx (hunk 3 above)
Comment (aappoalander) [reply src/core/Button/Button.tsx:21]: Fixed.

File: src/core/Button/Button.tsx (hunk 4 above)
Comment (aappoalander) [reply src/core/Button/Button.tsx:84]: Changed.

**Review by ketsappi — APPROVED** [review]

File: src/core/Button/Button.md (hunk 1 above)
Comment (aappoalander) [reply src/core/Button/Button.md]: As agreed, styling this will be handled later on when PAMU team is again available.

**Pull request merged**